cabal-install/Main.hs

```diff
@@ -268,6 +270,9 @@ mainWorker args = topHandler $
       , hiddenCmd  win32SelfUpgradeCommand win32SelfUpgradeAction
       , hiddenCmd  actAsSetupCommand actAsSetupAction
       , hiddenCmd  manpageCommand (manpageAction commandSpecs)
+
+      , hiddenCmd  installCommand { commandName = "new-configure" } newConfigureAction
```

Comment:
> Ah, OK.  But I think it is true that "new-build" is effectively "install"?

not currently; afaiui, it's more like a hypothetical `cabal build --auto-install-dependencies` with an implicit project-sandbox and a nix-style build-dep cache